src/main/java/ch/uzh/ifi/hase/soprafs21/controller/PathsApiController.java

```diff
@@ -54,4 +56,12 @@ public PathsApiController(NativeWebRequest request, UserService userService, Pat
         return ResponseEntity.status(HttpStatus.CREATED).build();
 
     }
+
+    @Override
+    public ResponseEntity<Void> deletePath(Long pathId) throws Exception {
+        UserOverviewDto authenticatedUser = (UserOverviewDto) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        pathService.deletePath(authenticatedUser.getId(), pathId);
```

Comment:
Please, move the authentication check to Service. We actually have the method for this already in UserService that you can use instead of duplicating the code.

Reply:
This will create unnecessary coupling between pathService and UserService

Reply:
https://softwareengineering.stackexchange.com/questions/230983/is-it-ok-for-services-to-reference-one-another

Reply:
Sure, Services can depend on each other but in this scenario it was avoidable. I have updated PR